libs/types-common/test/Test/Data/Mailbox.hs

```diff
@@ -0,0 +1,69 @@
+module Test.Data.Mailbox (tests) where
+
+import Data.ByteString.UTF8 qualified as UTF8BS
+import Data.Mailbox
+import Imports
+import Test.Tasty
+import Test.Tasty.HUnit
+import Text.Email.Parser
+
+validAddresses :: [(ByteString, Mailbox)]
+validAddresses =
+  [ ("john@doe.example", Mailbox Nothing $ unsafeEmailAddress "john" "doe.example"),
+    ("<john@doe.example>", Mailbox Nothing $ unsafeEmailAddress "john" "doe.example"),
+    ("John Doe<john@doe.example>", Mailbox (Just ["John", "Doe"]) $ unsafeEmailAddress "john" "doe.example"),
+    ("John Doe <john@doe.example>", Mailbox (Just ["John", "Doe"]) $ unsafeEmailAddress "john" "doe.example"),
+    ("John Q. Doe <john@doe.example>", Mailbox (Just ["John", "Q", ".", "Doe"]) $ unsafeEmailAddress "john" "doe.example"),
+    ("\"John Doe\" <john@doe.example>", Mailbox (Just ["John Doe"]) $ unsafeEmailAddress "john" "doe.example"),
+    ("\"John Doe\" (My Best Friend) <john@doe.example>", Mailbox (Just ["John Doe"]) $ unsafeEmailAddress "john" "doe.example"),
+    ("\"John@Doe.Example\" (My Friend @ Doe) <john@doe.example>", Mailbox (Just ["John@Doe.Example"]) $ unsafeEmailAddress "john" "doe.example"),
+    ("\"John    Doe\" (My Best Friend) <john@doe.example>", Mailbox (Just ["John    Doe"]) $ unsafeEmailAddress "john" "doe.example"),
+    ("\"John \\\"The J\\\" Doe\" <john@doe.example>", Mailbox (Just ["John \"The J\" Doe"]) $ unsafeEmailAddress "john" "doe.example"),
+    ("\"John not \\tab\" <john@doe.example>", Mailbox (Just ["John not tab"]) $ unsafeEmailAddress "john" "doe.example"),
```

Comment:
i don't understand this test.  shouldn't the parsed name be `John not \tab`?

Reply:
I think its to allow writing quotes here , for other characters, it should be ignored.